frontend/src/utilities/markdown.ts

```diff
@@ -2,14 +2,27 @@ import DOMPurify from 'dompurify';
 import { Converter } from 'showdown';
 
 export const markdownConverter = {
-  makeHtml: (markdown: string): string => {
-    const unsafeHtml = new Converter({
+  makeHtml: (markdown: string, maxHeading = 1): string => {
+    const converter = new Converter({
       tables: true,
       openLinksInNewWindow: true,
       strikethrough: true,
       emoji: true,
       literalMidWordUnderscores: true,
-    }).makeHtml(markdown);
+    });
+    let unsafeHtml = converter.makeHtml(markdown);
+
+    if (maxHeading > 1) {
+      // shift headings down (prevent them from competeing with page headings)
+      const shift = maxHeading - 1;
+      unsafeHtml = unsafeHtml.replace(
+        /<(\/?)h(\d)(.*?)>/gi,
+        (match, closingSlash, level, attributes) => {
+          const newLevel = Math.min(parseInt(level) + shift, 4); // cap at h4
```

Comment:
Why cap at h4? Aren't they valid up to h6?

Reply:
just to be in-line with the current `ALLOWED_TAGS` in that file